sphinx/ext/autosummary/generate.py

```diff
@@ -630,6 +645,10 @@ def get_parser() -> argparse.ArgumentParser:
                         dest='imported_members', default=False,
                         help=__('document imported members (default: '
                                 '%(default)s)'))
+    parser.add_argument('-a', '--respect-__all__', action='store_false',
```

Comment:
+1 for the word "respect". So this will be renamed to `--respect-module-all`.
Please add an entry for this option into `doc/man/sphinx-autogen.rst` and CHANGES.